### PR TITLE
Leads

### PR DESCRIPTION
**Pull request by Bekmirzo**
Added Leads domain

### CODE REVIEW

File: src/domain/leads/dto/leads.dto.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    readonly tg_username: string;
+
+    @IsString()
+    readonly phone: string;
+
+    @IsString()
+    readonly course_id: string;
+}
+
+export class LeadDto extends CreateLeadDto {
```
Comment (azoirov) [inline src/domain/leads/dto/leads.dto.ts]: buni ochirib tasshlayvering

File: src/domain/leads/leads.repo.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+class leadRepo {
+  public leadModel = leadModel;
+
+  create = async (data: CreateLeadDto): Promise<CreateLeadDto> => {
+    return this.leadModel.create(data);
+  };
+
+
+  getAll = async (): Promise<LeadDto[]> => {
```
Comment (azoirov) [inline src/domain/leads/leads.repo.ts]: bu joylarda interface bilan check qilsa boladi hammasida

**Review by azoirov — CHANGES_REQUESTED** [review]

File: src/domain/leads/dto/leads.dto.ts (hunk 1 above)
Comment (Bekmirzo) [reply src/domain/leads/dto/leads.dto.ts]: Bu yerda **CreateLeadDto**  o'chirilib faqat **LeadDto** qoldirildi

File: src/domain/leads/leads.repo.ts (hunk 2 above)
Comment (Bekmirzo) [reply src/domain/leads/leads.repo.ts]: Repodagi **getAll** va **create** funksiyalari interface bilan tekshiriladigan qilindi:
`create = async (data: Ilead): Promise<Ilead> => {
        return this.leadModel.create(data);
    };
`
`
    getAll = async (): Promise<Ilead[]> => {
        return this.leadModel.find().lean();
    };
`

**Pull request merged**